components/hab/src/command/studio/enter.rs

```diff
@@ -13,17 +13,55 @@ use crate::{config,
 
 pub const ARTIFACT_PATH_ENVVAR: &str = "ARTIFACT_PATH";
 
+const AUTH_TOKEN_ENVVAR: &str = "HAB_AUTH_TOKEN";
+const BLDR_URL_ENVVAR: &str = "HAB_BLDR_URL";
+const CTL_SECRET_ENVVAR: &str= "HAB_CTL_SECRET";
 const ORIGIN_ENVVAR: &str = "HAB_ORIGIN";
 const STUDIO_CMD: &str = "hab-studio";
 const STUDIO_CMD_ENVVAR: &str = "HAB_STUDIO_BINARY";
 const STUDIO_PACKAGE_IDENT: &str = "core/hab-studio";
 
+// fn set_env_var(env_var: &str) -> Result<()> {
+//   let config = config::load()?;
+//   if henv::var(env_var).is_err() {
+//     if let Some(config_val) = match env_var {
```

Comment:
Maybe I don't understand what `config.bldr_url` is doing...  it's not a function call, right? (`config.bldr_url()`)

I want to set `config_val` based on `config.<something>` depending on `env_var`...  I know how I'd do it in ruby...

The other version might be the most elegant?

Reply:
`config.bldr_url` is an `Option<String>` value defined in [hab/src/config.rs](https://github.com/habitat-sh/habitat/blob/master/components/hab/src/config.rs#L23). I'm not quite sure I understand the rest of your question. Does knowing the type explain it?

Reply:
Ah ha!  And `Config` is a struct so `.bldr_url` is a... "property"(?).  :+1:

Reply:
I believe the common term in rust is ["field"](https://doc.rust-lang.org/rust-by-example/custom_types/structs.html)